datafusion/physical-plan/src/sorts/sort.rs

```diff
@@ -1224,6 +1245,28 @@ impl ExecutionPlan for SortExec {
                 .with_preserve_partitioning(self.preserve_partitioning()),
         )))
     }
+
+    // Pass though filter pushdown.
+    // This often happens in partitioned plans with a TopK because we end up with 1 TopK per partition + a final TopK at the end.
+    // Implementing this pass-through allows global/top/final TopK to push down filters to the partitions.
+    fn push_down_filter(
+        &self,
+        expr: Arc<dyn PhysicalExpr>,
+    ) -> Result<Option<Arc<dyn ExecutionPlan>>> {
```

Comment:
This won't do anything for the multi-partition case _until_ we implement filter pushdown for `SortingMergeExec`

Reply:
https://github.com/apache/datafusion/issues/15534